Sources/BraintreeThreeDSecure/BTThreeDSecureClient.swift

```diff
@@ -139,14 +197,112 @@ extension BTPaymentFlowClient {
             }
         }
     }
+    
+    // MARK: - Private Methods
+    
+    /// Prepare for a 3DS 2.0 flow.
+    /// - Parameters:
+    ///   - apiClient: The API client.
+    ///   - completion: This completion will be invoked exactly once. If the error is nil then the preparation was successful.
+    private func prepareLookup(
+        request: BTThreeDSecureRequest,
+        completion: @escaping (Error?) -> Void
+    ) {
+        apiClient.fetchOrReturnRemoteConfiguration { [weak self] configuration, error in
+            guard let self else { return }
+
+            guard let configuration, error == nil else {
+                completion(error)
+                return
+            }
+
+            if configuration.cardinalAuthenticationJWT != nil {
+                self.threeDSecureV2Provider = BTThreeDSecureV2Provider(
+                    configuration: configuration,
+                    apiClient: self.apiClient,
+                    request: request
+                ) { lookupParameters in
+                    if let dfReferenceID = lookupParameters?["dfReferenceId"] {
+                        request.dfReferenceID = dfReferenceID
+                    }
+                    completion(nil)
+                }
+            } else {
+                completion(BTThreeDSecureError.configuration("Merchant is not configured for 3SD 2."))
+                return
+            }
+        }
+    }
+        
+    private func start(request: BTThreeDSecureRequest, configuration: BTConfiguration) {
```

Comment:
Nice and clean now!

Reply:
Wow Sammy, great job on this!

Reply:
We can definitely further simplify internally after the GA release, too!